githubWebClient.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+var USERNAME = process.env.GITHUB_USER;
+var PASSWORD = process.env.GITHUB_PASSWORD;
+var cheerio = require('cheerio');
+var childProcess = require('child_process');
+var CookieJar = require('./cookieJar');
+var jar = new CookieJar();
+
+/**
+ * Scrape github login page
+ * @return {Object} Object containeing two keys: headers (a string) and body (also a string)
+ */
+var githubLogin = function() {
+  var output = childProcess.execSync('curl -v -L https://github.com/login 2>&1').toString().split('<!DOCTYPE html>');
+  return {
+    headers: output[0],
+    body: output[1]    
+  };
+};
+
+
+/**
+ * gets a CSRF token by hitting github's login form
+ * @return {String} CSRF token
+ */
+var getAuthenticityToken = function() {
+  var login = githubLogin();
+  jar.parseHeaders(login.headers);
+  return encodeURIComponent(cheerio.load(login.body)('[name=authenticity_token]').val());
```

Comment:
It feel very heavyweight to npm install cheerio and parse the entire dom just for this. Could you instead extract this via a regex:

```javascript
login.body.match(/name="authenticity_token".*value="([^"]+)"/)[1]
```